src/components/MoneyRequestAmountInput.tsx

```diff
@@ -0,0 +1,251 @@
+
+    const [selection, setSelection] = useState({
+        start: selectedAmountAsString.length,
+        end: selectedAmountAsString.length,
+    });
+
+    const forwardDeletePressedRef = useRef(false);
+
+    /**
+     * Sets the selection and the amount accordingly to the value passed to the input
+     * @param {String} newAmount - Changed amount from user input
+     */
+    const setNewAmount = useCallback(
+        (newAmount: string) => {
+            // Remove spaces from the newAmount value because Safari on iOS adds spaces when pasting a copied value
+            // More info: https://github.com/Expensify/App/issues/16974
+            const newAmountWithoutSpaces = MoneyRequestUtils.stripSpacesFromAmount(newAmount);
+            // Use a shallow copy of selection to trigger setSelection
+            // More info: https://github.com/Expensify/App/issues/16385
+            if (!MoneyRequestUtils.validateAmount(newAmountWithoutSpaces, decimals)) {
+                setSelection((prevSelection) => ({...prevSelection}));
+                return;
+            }
+
+            // setCurrentAmount contains another setState(setSelection) making it error-prone since it is leading to setSelection being called twice for a single setCurrentAmount call. This solution introducing the hasSelectionBeenSet flag was chosen for its simplicity and lower risk of future errors https://github.com/Expensify/App/issues/23300#issuecomment-1766314724.
+
+            let hasSelectionBeenSet = false;
+            setCurrentAmount((prevAmount) => {
+                const strippedAmount = MoneyRequestUtils.stripCommaFromAmount(newAmountWithoutSpaces);
+                const isForwardDelete = prevAmount.length > strippedAmount.length && forwardDeletePressedRef.current;
+                if (!hasSelectionBeenSet) {
+                    hasSelectionBeenSet = true;
+                    setSelection((prevSelection) => getNewSelection(prevSelection, isForwardDelete ? strippedAmount.length : prevAmount.length, strippedAmount.length));
+                }
+                onAmountChange?.(strippedAmount);
+                return strippedAmount;
+            });
+        },
+        [decimals, onAmountChange],
+    );
+
+    useImperativeHandle(moneyRequestAmountInputRef, () => ({
+        setNewAmount(amountValue: string) {
+            setNewAmount(amountValue);
+        },
+        changeSelection(newSelection: Selection) {
+            setSelection(newSelection);
+        },
+        changeAmount(newAmount: string) {
+            setCurrentAmount(newAmount);
+        },
+        getAmount() {
+            return currentAmount;
+        },
+        getSelection() {
+            return selection;
+        },
+    }));
+
+    useEffect(() => {
+        if (!currency || typeof amount !== 'number') {
+            return;
+        }
+        const frontendAmount = amount ? CurrencyUtils.convertToFrontendAmount(amount).toString() : '';
+        setNewAmount(frontendAmount);
```

Comment:
```suggestion
        setCurrentAmount(frontendAmount);
```

We need to use `setCurrentAmount` here as used previously. We shouldn't be triggering `onAmountChange` from here because that's supposed to be only triggered via user input and via `setNewAmount`

Reply:
Nice catch on this one 👍, I should have caught it earlier 🤦